src/main/java/org/snomed/snowstorm/core/data/domain/QueryConcept.java

```diff
@@ -299,10 +300,22 @@ private static String serializeMap(Map<Integer, Map<String, List<Object>>> group
 					builder.append(type);
 					builder.append("=");
 					for (Object value : attributes.get(type)) {
-						if (!(value instanceof String)) {
+						String valueString;
+						if (value instanceof String) {
+							valueString = (String) value;
+						} else {
 							builder.append("#");
+							if (value instanceof Double || value instanceof Float) {
+								// Maximum decimal places is 6 - See https://confluence.ihtsdotools.org/display/mag/Concrete+Domain+Decimal+Places+and+Rounding
+								valueString = new DecimalFormat("#0.0#####").format(value);
+							} else {
+								valueString = value.toString();
+							}
 						}
-						builder.append(value);
+						if (valueString.contains(".")) {
+							System.out.println();
+						}
```

Comment:
What's this block for @kaicode? It looks like a left over from some testing?

Reply:
This is left over from debugging. Will remove.